db/records/operations/group.py

```diff
@@ -1,123 +1,240 @@
-from sqlalchemy import select, Column, func
+from enum import Enum
+import json
+import logging
+from sqlalchemy import select, func, and_, case, literal
 
-from db.records.exceptions import BadGroupFormat, GroupFieldNotFound
-from db.records.operations.select import get_query, apply_filters
+from db.records import exceptions as rec_exc
```

Comment:
I think it would be clearer to have a non-abbreviated name here e.g. `records_exceptions`